src/context.rs

```diff
@@ -0,0 +1,103 @@
+use std::collections::HashMap;
+
+use serde::Serialize;
+use windows_core::{BSTR, VARIANT};
+
+use crate::{WMIConnection, WMIResult};
+
+#[derive(Debug, PartialEq, Serialize, Clone)]
+#[serde(untagged)]
+pub enum ContextValueType {
+    String(String),
+    I4(i32),
+    R8(f64),
+    Bool(bool),
+}
+
+impl From<ContextValueType> for VARIANT {
+    fn from(value: ContextValueType) -> Self {
+        match value {
+            ContextValueType::Bool(b) => Self::from(b),
+            ContextValueType::I4(i4) => Self::from(i4),
+            ContextValueType::R8(r8) => Self::from(r8),
+            ContextValueType::String(str) => Self::from(BSTR::from(str)),
+        }
+    }
+}
+
+impl WMIConnection {
```

Comment:
I think it'll be better to try and keep the API closer to the native one, maybe something like
```
struct WMIContext(IWbemContext);

impl WMIConnection() {
    pub fn ctx() -> &mut WMIContext { .. }
}

impl WMIContext {
   pub fn set_value(&mut self, key: &str, value: impl Into<ContextValueType>) -> WMIResult<()> { ... }
   pub fn delete_all(..) { ... }
}

// Usage should be
// connection.ctx().set_value("IncludeHidden", true)?;
```

Reply:
Ah nice, this is much better. I have updated based on the suggestion, thanks!